core/src/test/scala/unit/kafka/server/KafkaMetricReporterClusterIdTest.scala

```diff
@@ -75,44 +80,34 @@ object KafkaMetricReporterClusterIdTest {
   }
 }
 
-class KafkaMetricReporterClusterIdTest extends QuorumTestHarness {
-  var server: KafkaServer = _
-  var config: KafkaConfig = _
-
-  @BeforeEach
-  override def setUp(testInfo: TestInfo): Unit = {
-    super.setUp(testInfo)
-    val props = TestUtils.createBrokerConfig(1, zkConnect)
-    props.setProperty(KafkaConfig.KafkaMetricsReporterClassesProp, "kafka.server.KafkaMetricReporterClusterIdTest$MockKafkaMetricsReporter")
-    props.setProperty(KafkaConfig.MetricReporterClassesProp, "kafka.server.KafkaMetricReporterClusterIdTest$MockBrokerMetricsReporter")
-    props.setProperty(KafkaConfig.BrokerIdGenerationEnableProp, "true")
-    props.setProperty(KafkaConfig.BrokerIdProp, "-1")
-    config = KafkaConfig.fromProps(props)
-    server = new KafkaServer(config, threadNamePrefix = Option(this.getClass.getName))
-    server.startup()
+class KafkaMetricReporterClusterIdTest extends KafkaServerTestHarness {
+
+  def generateConfigs: Seq[KafkaConfig] = {
+    val overridingProps = new Properties
+    overridingProps.setProperty(KafkaConfig.KafkaMetricsReporterClassesProp, "kafka.server.KafkaMetricReporterClusterIdTest$MockKafkaMetricsReporter")
+    overridingProps.setProperty(KafkaConfig.MetricReporterClassesProp, "kafka.server.KafkaMetricReporterClusterIdTest$MockBrokerMetricsReporter")
+    overridingProps.setProperty(KafkaConfig.BrokerIdGenerationEnableProp, "true")
+    TestUtils.createBrokerConfigs(1, zkConnectOrNull).
+      map(KafkaConfig.fromProps(_, overridingProps))
   }
 
-  @Test
-  def testClusterIdPresent(): Unit = {
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumName)
+  @ValueSource(strings = Array("zk","kraft"))
+  def testClusterIdPresent(quorum: String): Unit = {
     assertEquals("", KafkaMetricReporterClusterIdTest.setupError.get())
 
-    assertNotNull(KafkaMetricReporterClusterIdTest.MockKafkaMetricsReporter.CLUSTER_META)
-    isValidClusterId(KafkaMetricReporterClusterIdTest.MockKafkaMetricsReporter.CLUSTER_META.get().clusterId())
-
     assertNotNull(KafkaMetricReporterClusterIdTest.MockBrokerMetricsReporter.CLUSTER_META)
     isValidClusterId(KafkaMetricReporterClusterIdTest.MockBrokerMetricsReporter.CLUSTER_META.get().clusterId())
 
-    assertEquals(KafkaMetricReporterClusterIdTest.MockKafkaMetricsReporter.CLUSTER_META.get().clusterId(),
-      KafkaMetricReporterClusterIdTest.MockBrokerMetricsReporter.CLUSTER_META.get().clusterId())
+    if(!isKRaftTest()) {
```

Comment:
Thanks for the review. `MockKafkaMetricsReporter` is not preset in the `BrokerServer` that is started in case of Kraft. That class is part of `KafkaServer` which is only started up in case of Kafka with ZK, as per the code in `KafkaServerTestHarness`.